src/libraries/System.Runtime.Serialization.BinaryFormat/src/System/Runtime/Serialization/BinaryFormat/SerializationRecord.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+/// <summary>
+/// Abstract class that represents the serialization record.
+/// </summary>
+/// <remarks>
+///  <para>
+///   Every instance returned to the end user can be either <see cref="PrimitiveTypeRecord{T}"/>,
+///   a <see cref="ClassRecord"/> or an <see cref="ArrayRecord"/>.
+///  </para>
+/// </remarks>
+[DebuggerDisplay("{RecordType}, {ObjectId}")]
+#if SYSTEM_RUNTIME_SERIALIZATION_BINARYFORMAT
+public
+#else
+internal
+#endif
+abstract class SerializationRecord
+{
+    internal const int NoId = 0;
+
+    internal SerializationRecord() // others can't derive from this type
+    {
+    }
+
+    /// <summary>
+    /// Gets the type of the record.
+    /// </summary>
+    /// <value>The type of the record.</value>
+    public abstract RecordType RecordType { get; }
+
+    /// <summary>
+    /// Gets the ID of the record.
+    /// </summary>
+    /// <value>The ID of the record.</value>
+    public abstract int ObjectId { get; }
+
+    /// <summary>
+    /// Compares the type and assembly name read from the payload against the specified type.
+    /// </summary>
+    /// <remarks>
+    /// <para>This method takes type forwarding into account.</para>
+    /// <para>This method does NOT take into account member names or their types.</para>
+    /// </remarks>
+    /// <param name="type">The type to compare against.</param>
+    /// <returns>True if the serialized type and assembly name match provided type.</returns>
```

Comment:
```suggestion
    /// <returns><see langword="true" /> if the serialized type and assembly name match provided type; otherwise, <see langword="false" />.</returns>
```